pkg/middlewares/vpc_security_group_rules_test.go

```diff
@@ -14,61 +13,111 @@ import (
 func TestVPCSecurityGroupRuleSanitizer(t *testing.T) {
 
 	factory := &terraform.MockResourceFactory{}
-	factory.On("CreateResource", mock.Anything, "aws_security_group_rule").Times(8).Return(nil, nil)
+	factory.On("CreateAbstractResource", aws.AwsSecurityGroupRuleResourceType, "sgrule-1175318309", mock.Anything).Times(1).Return(
```

Comment:
Just to understand, you use `mock.Anything` because if any of the fields does not match the hash will change and thus the mock will not match anymore ? Maybe it will be easier to maintain to use `mock.Anything` on the ID and assert the whole attribute map ?

Reply:
the test use to no check anything I added a check on the ID as it is generated on the field and it was easier to implement than testing with the whole attribute map...

Reply:
🆗